source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -1646,5 +1729,186 @@ namespace RendererQml
+            const bool isShowCardButton = Utils::IsInstanceOfSmart<AdaptiveCards::ShowCardAction>(action);
+            const bool isIconLeftOfTitle = actionsConfig.iconPlacement == AdaptiveCards::IconPlacement::LeftOfTitle;
+            std::shared_ptr<QmlTag> showCardIcon;
+
+            auto buttonElement = std::make_shared<QmlTag>("Button");
+            buttonElement->Property("id", buttonId);
+
+            if (isShowCardButton)
+            {
+                buttonElement->Property("property bool showCard", "false");
+            }
+
+            //Add button background
+            auto bgRectangle = std::make_shared<QmlTag>("Rectangle");
+            bgRectangle->Property("id", Formatter() << buttonId << "_bg");
+            bgRectangle->Property("anchors.fill", "parent");
+            bgRectangle->Property("radius", Formatter() << buttonId << ".height / 2");
+            bgRectangle->Property("border.width", "1");                                   
+
+            //Add button content item
+            auto contentItem = std::make_shared<QmlTag>("Item");
+            auto contentLayout = std::make_shared<QmlTag>(isIconLeftOfTitle ? "Row" : "Column");
+            contentLayout->Property("id", Formatter() << buttonId << (isIconLeftOfTitle ? "_row" : "_col"));
+            contentLayout->Property("spacing", "5");
+            contentLayout->Property("leftPadding", "5");
+            contentLayout->Property("rightPadding", "5");
+
+            contentItem->AddChild(contentLayout);
+            contentItem->Property("implicitHeight", Formatter() << contentLayout->GetId() << ".implicitHeight");
+            contentItem->Property("implicitWidth", Formatter() << contentLayout->GetId() << ".implicitWidth");
+
+            //Add button icon
+            if (!action->GetIconUrl().empty())
+            {
+                auto contentImage = std::make_shared<QmlTag>("Image");
+                contentImage->Property("id", Formatter() << buttonId << "_img");
+                contentImage->Property("height", Formatter() << fontSize);
+                contentImage->Property("width", Formatter() << fontSize);
+                contentImage->Property("fillMode", "Image.PreserveAspectFit");
+
+                if (isIconLeftOfTitle)
+                {
+                    contentImage->Property("anchors.verticalCenter", "parent.verticalCenter");
+                }
+                else
+                {
+                    contentImage->Property("anchors.horizontalCenter", "parent.horizontalCenter");
+                }
+
+                //TODO: Adding dummy image. This should be replaced!
+                std::string file_path = __FILE__;
+                std::string dir_path = file_path.substr(0, file_path.rfind("\\"));
+                dir_path.append("\\Images\\Cat.png");
+                std::replace(dir_path.begin(), dir_path.end(), '\\', '/');
+                contentImage->Property("source", "\"" + std::string("file:/") + dir_path + "\"");
```

Comment:
This is repetitive. We should find a better solution to call image file. I will work on a separate PR to try get rid of this.

Reply:
should we convert it into function where we pass in the file name as argument and the function returns the complete path !!

Reply:
The idea is to remove this permanently and use a qrc instead